src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,62 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String FIRST_PARAMETER = "supply";
+    private static final String SECOND_PARAMETER = "buy";
+    private static final String THIRD_PARAMETER = "result";
+    private static final String SEPARATOR = ",";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String stringsFromFile = readFromFile(fromFileName);
+        String report = createReport(stringsFromFile);
+        writeReport(report, toFileName);
+    }
+
+    private String readFromFile(String fileName) {
+        StringBuilder stringBuilder = new StringBuilder();
+        try {
+            BufferedReader bufferedReader = new BufferedReader(new FileReader(fileName));
+
+            String value = bufferedReader.readLine();
+            while (value != null) {
+                stringBuilder.append(value).append(';');
+                value = bufferedReader.readLine();
+            }
+            return stringBuilder.toString();
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file " + fileName + ".", e);
+        }
+    }
+
+    private String createReport(String readFromFile) {
+        StringBuilder stringBuilder = new StringBuilder();
+        String[] splitter = readFromFile.split(";");
+        int supplySum = 0;
+        int buySum = 0;
+        for (String element : splitter) {
+            supplySum += element.charAt(0) == FIRST_PARAMETER.charAt(0) ? Integer.parseInt(element
+                    .substring(FIRST_PARAMETER.length() + 1)) : 0;
+            buySum += element.charAt(0) == SECOND_PARAMETER.charAt(0) ? Integer.parseInt(element
+                    .substring(SECOND_PARAMETER.length() + 1)) : 0;
```

Comment:
Do you really need to try both cases? Maybe it is better to use if-else construction here?